proto/src/determined/trial/v1/trial.proto

```diff
@@ -144,6 +144,8 @@ message Trial {
   repeated string task_ids = 20;
   // Signed searcher metrics value.
   double searcher_metric_value = 21;
+  // Number of days to retain logs for.
+  int32 log_retention_days = 22;
```

Comment:
qq: even if `log_retention_days` in DB is `null`, trial API returns `log_retention_days: 0`. it used to return `null`. Is it expected? 
For example, didn't set retention days for this trials, but show `0` by default. 
Also, when `log_retention_days` is `null`, does that mean `log_retention_days` is `forever` or `0`?

<img width="173" alt="image" src="https://github.com/determined-ai/determined/assets/109113616/3fa9cbd7-de3f-4ff2-a400-7d96cb30adf5">